lib/active_public_resources/drivers/vimeo.rb

```diff
@@ -84,16 +80,26 @@ def perform_request(request_criteria)
         request_criteria.validate_presence!([:query])
         raise StandardError.new("driver has not been initialized properly") unless @client
 
-        results = @client.search(request_criteria.query, {
-          :page           => request_criteria.page || 1,
-          :per_page       => request_criteria.per_page || 25,
-          :full_response  => 1,
-          :sort           => request_criteria.sort || "relevant",
-          :user_id        => nil,
-          :content_filter => request_criteria.content_filter || 'safe'
-        })
+        results = HTTParty.get('https://api.vimeo.com/videos',
+          query: {
+            query: request_criteria.query,
+            page: request_criteria.page || 1,
+            per_page: request_criteria.per_page || 25,
+            sort: 'relevant',
+            filter: 'content_rating',
+            filter_content_rating: 'safe'
+          },
+          headers: { "Authorization" => "Bearer #{@access_token}" }
+        )
 
-        return parse_results(request_criteria, results)
+        if results.code == 401
```

Comment:
If you want to get rid of the nested `if` and the `else` I think we could do something like this:
```
return parse_results(request_criteria, JSON.parse(results)) unless results.code == 401
if !@client.verify_token?(@access_token)
  @access_token = @client.get_access_token
  perform_request(request_criteria) if @access_token
end
```